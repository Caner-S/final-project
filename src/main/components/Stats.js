import React from 'react';
import { getUnavailableBookings} from "../dao/BookingDao";

class Stats extends React.Component {
    constructor() {
        super();
        let today = new Date().toISOString().slice(0, 10);
        this.state = {
            bookings: [],
            todayISO: today+"T23:59",
            arrivalDate: today+"T09:00",
            departureDate: today+"T17:00",
        };
    }

    componentDidMount() {
        let currentComponent = this;
        getUnavailableBookings(this.state.arrivalDate, this.state.departureDate, this.state.todayISO).get().then(function(querySnapshot) {
            querySnapshot.forEach(function(doc) {
                let updatedData = querySnapshot.docs.map(doc => doc)
                currentComponent.setState({ bookings: updatedData })

            })

        })

    }



    render() {
        return (
            <div>
            <h1>Stats</h1>
            <h2>Total Bookings Today</h2>
                <h1>{((this.state.bookings.length/209)*100).toFixed(2)}%</h1>
                <h2>Area Breakdown</h2>
            </div>
        );
    }
}
export default Stats;