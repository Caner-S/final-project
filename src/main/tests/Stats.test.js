import React from 'react';
import renderer from 'react-test-renderer';

import Stats from '../components/Stats';

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<Stats />).toJSON();
    expect(tree).toMatchSnapshot();
});