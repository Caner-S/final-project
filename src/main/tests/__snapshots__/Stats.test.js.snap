// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <h1>
    Stats
  </h1>
  <h2>
    Total Bookings Today
  </h2>
  <h1>
    0.00
    %
  </h1>
  <h2>
    Area Breakdown
  </h2>
</div>
`;
